### PR TITLE
fix : Information lost on switching between login/signup fragments

### LINKED ISSUE #608: Information lost on repetitive clicking on SignUp button on navigation bar
**What is the issue about ?**
I'm not sure whether this is considered as an issue or not, please look into it. 
When we fill credentials in signup fragment and click on SignUp button in bottom 
navigation bar, all the filled information is lost. 

**Screen-Shots**
whatsapp image 2018-10-01 at 10 39 21 pm

**Would you like to work on this issue?**
Yes (I want mentors to approve this issue so that I can start working on it!)

### PR DESCRIPTION
**Pull request by shikhart98**
Fixes #608

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.2.0-alpha16'
+        classpath 'com.android.tools.build:gradle:3.2.1'
```
Comment (iamareebjamal) [inline build.gradle]: Unrelated change

File: app/src/main/java/org/fossasia/openevent/general/AuthActivity.kt [hunk 2]
```diff
@@ -43,14 +44,25 @@ class AuthActivity : AppCompatActivity() {
 
         supportActionBar?.title = "Login"
 
-        loadFragment(LoginFragment())
+        checkAndLoadFragment(LoginFragment())
+    }
+
+    private fun checkAndLoadFragment(fragment: Fragment) {
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/AuthActivity.kt]: @iamareebjamal should we move `checkAndLoadFrag` and `loadFragment` to the utility class?

File: build.gradle (hunk 1 above)
Comment (shikhart98) [reply build.gradle]: I think gradle did that! I'll fix it.

File: app/src/main/java/org/fossasia/openevent/general/AuthActivity.kt (hunk 2 above)
Comment (shikhart98) [reply app/src/main/java/org/fossasia/openevent/general/AuthActivity.kt]: untitled
it is showing this!

Comment (shikhart98) [reply app/src/main/java/org/fossasia/openevent/general/AuthActivity.kt]: I'll still change it to previous version and add a commit.

**Comment by iamareebjamal** [comment]
> should we move checkAndLoadFrag and loadFragment to the utility class
> Yes

**Comment by iamareebjamal** [comment]
> should we move checkAndLoadFrag and loadFragment to the utility class
> Yes

**Comment by iamareebjamal** [comment]
> should we move checkAndLoadFrag and loadFragment to the utility class
> Yes

**Comment by iamareebjamal** [comment]
> should we move checkAndLoadFrag and loadFragment to the utility class
> Yes

**Comment by iamareebjamal** [comment]
> should we move checkAndLoadFrag and loadFragment to the utility class
> Yes

**Comment by shikhart98** [comment]
@iamareebjamal is there somthing else I have to do in here?

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/open-event-android/pull/646#issuecomment-431744284

**Comment by shikhart98** [comment]
@iamareebjamal  I ahve updated Utils with checkandloadfragment() and loadfragment() and added a new parameter containerID to make it generic to use with other containers as well. Hence our MainActivity and AuthActivity is perfectly synced with these methods in Utils.

**Comment by shikhart98** [comment]
@iamareebjamal please review

File: app/src/main/java/org/fossasia/openevent/general/AuthActivity.kt [hunk 3]
```diff
@@ -37,33 +39,14 @@ class AuthActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_auth)
         if (this.intent.extras != null) {
-            bundle = this.intent.extras
+            Utils.bundle = this.intent.extras
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/AuthActivity.kt]: So bad! Why?

**Comment by simarsingh24** [comment]
closing this since there has been no activity from the last two months on this @shikhart98 please raise another PR for the same with the necessary changes!
